src/Domain/ValueObjects/Wallet.cs

```diff
@@ -0,0 +1,29 @@
+using Domain.Aggregates;
+
+namespace Domain.ValueObjects;
+
+public record Wallet(decimal Balance)
+{
+
+    public decimal Balance { get; private set; } = Balance;
```

Comment:
we could also add a check here, so that balance is asserted not negative during initialization
```cs
public decimal Balance { get; private set; } = Balance >= 0 ? Balance : throw new ArgumentException(nameof(Balance), "Wallet balance cannot be negative!");